googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java

```diff
@@ -120,8 +121,13 @@ public File call(final AbstractHttpEntity entity) throws BackgroundException {
                         switch(postResponse.getStatusLine().getStatusCode()) {
                             case HttpStatus.SC_OK:
                                 if(status.isExists()) {
-                                    return session.getClient().getObjectParser().parseAndClose(
+                                    final File f = session.getClient().getObjectParser().parseAndClose(
                                             new InputStreamReader(postResponse.getEntity().getContent(), StandardCharsets.UTF_8), File.class);
+                                    if(null != status.getTimestamp()) {
+                                        new DriveTimestampFeature(session, fileid).setTimestamp(file, status.getTimestamp());
```

Comment:
I would duplicate the implementation which allows to directly return the updated `File` reference.

```suggestion
            final File properties = new File();
            properties.setModifiedTime(new DateTime(status.getTimestamp()));
            return session.getClient().files().update(fileid, properties).setFields(DriveAttributesFinderFeature.DEFAULT_FIELDS).
                    setSupportsAllDrives(new HostPreferences(session.getHost()).getBoolean("googledrive.teamdrive.enable")).execute();
```

Reply:
Create implementation method in Timestamp-feature that can be used from both Timestamp-feature and Write-feature, without duplicating code.

Reply:
The `update` method must not contain fields that are not modifiable. The suggested change thus leads to 403 `The resource body includes fields which are not directly writable`.